torch_xla/csrc/tensor.cpp

```diff
@@ -607,6 +607,17 @@ bool XLATensor::ShouldSyncIrNode() {
   return this->data()->ir_value->op() != xla_device_data;
 }
 
+bool XLASymNodeImpl::is_bool() {
+  auto op = node()->op().op;
+  // Reference:
+  // https://github.com/pytorch/pytorch/blob/master/torch/fx/experimental/symbolic_shapes.py#L403
+  if (op == at::aten::eq || op == at::aten::ne || op == at::aten::ge ||
+      op == at::aten::lt) {
```

Comment:
if you support boolean negation, that'll be good too. This is a weird way to do it but absent proper type propagation it's probably ok.

Reply:
Yea, otherwise I will need to type check the object which is probably weird too.